From 772d58c2a5d7247725a4fdec0f232c6b24327343 Mon Sep 17 00:00:00 2001
From: yao zhang1 <yao.zhang1@amlogic.com>
Date: Tue, 7 Feb 2023 14:57:42 +0800
Subject: [PATCH 91/95] build: C3 do not check the start of physical memory.
 [1/1]

PD#SWPL-106998

Problem:
Boottime is too slow.

Solution:
Speed up boottime 50ms.

Verify:
C3 aw402

Change-Id: I48d8d319f933d89fa9c32c91db7139c51c47ce9d
Signed-off-by: yao zhang1 <yao.zhang1@amlogic.com>
---
 arch/arm/boot/compressed/fdt_check_mem_start.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/boot/compressed/fdt_check_mem_start.c b/arch/arm/boot/compressed/fdt_check_mem_start.c
index 62450d824c3c..1df20cc445cf 100644
--- a/arch/arm/boot/compressed/fdt_check_mem_start.c
+++ b/arch/arm/boot/compressed/fdt_check_mem_start.c
@@ -68,6 +68,11 @@ uint32_t fdt_check_mem_start(uint32_t mem_start, const void *fdt)
 	const char *type;
 	int offset, len;
 
+#if IS_ENABLED(CONFIG_AMLOGIC_LINUX_S_OPTIMIZE)
+	//Optimize for boottime
+	if (fdt)
+		return mem_start;
+#endif
 	if (!fdt)
 		return mem_start;
 
-- 
2.25.1

